e2e/nomostest/gitproviders/git-provider.go

```diff
@@ -63,3 +63,29 @@ func NewGitProvider(t testing.NTB, provider string, logger *testlogger.TestLogge
 		return &LocalProvider{}
 	}
 }
+
+// WithLogger updates an existing GitProvider to use the provider TestLogger
+func WithLogger(t testing.NTB, provider GitProvider, logger *testlogger.TestLogger) GitProvider {
+	switch p := provider.(type) {
+	case *BitbucketClient:
+		return &BitbucketClient{
+			oauthKey:     p.oauthKey,
+			oauthSecret:  p.oauthSecret,
+			refreshToken: p.refreshToken,
+			logger:       logger,
+		}
+	case *GitlabClient:
+		return &GitlabClient{
+			privateToken: p.privateToken,
+		}
+	case *LocalProvider:
+		// The PortForwarder for LocalProvider is reset for each test in setupTestCase.
+		// Therefore, we don't need to reset the PortForwarder's logger here.
+		return &LocalProvider{
+			PortForwarder: p.PortForwarder,
+		}
+	default:
+		t.Fatalf("unrecognized GitProvider: %v", p)
+	}
+	return nil
```

Comment:
this is unreachable. if you need it to compile, move it to the default after the fatal

Reply:
yeah it's just to compile. I can move it to the default, figured it doesn't make much difference

Reply:
done